### ISSUE TITLE
Dart Analyzer Error Encountered a Dart Analysis Server error.

### ISSUE THREAD

**Issue by zzzmyfox** [dart-lang/sdk#38208]
Analyzer Feedback from IntelliJ

## Version information

- `IDEA AI-191.8026.42.35.5791312`
- `2.2.0-edge.0a7dcf17eb5f2450480527d6ad1e201fb47f1e36`
- `AI-191.8026.42.35.5791312, JRE 1.8.0_202-release-1483-b49-5587405x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.14.6, screens 1920x1080`

## Exception
```
Dart analysis server, SDK version 2.2.0-edge.0a7dcf17eb5f2450480527d6ad1e201fb47f1e36, server version 1.23.0, error: Failed to handle completion domain request: {id: 2025, method: completion.getSuggestionDetails, params: {file: /Users/zzex/Desktop/jyxt_flutter_app/lib/ui/pages/web_view_page.dart, id: 737, label: Key, offset: 198}, clientRequestTime: 1567668700520}
UnimplementedError: (FieldFormalParameterImpl) @required this.url
#0      SyntacticScopeNamesCollector._addFormalParameter (package:analyzer_plugin/src/utilities/change_builder/dart/syntactic_scope.dart:275:7)
#1      SyntacticScopeNamesCollector._addFormalParameter (package:analyzer_plugin/src/utilities/change_builder/dart/syntactic_scope.dart:265:7)
#2      SyntacticScopeNamesCollector._addFormalParameters (package:analyzer_plugin/src/utilities/change_builder/dart/syntactic_scope.dart:281:7)
#3      SyntacticScopeNamesCollector.visitConstructorDeclaration (package:analyzer_plugin/src/utilities/change_builder/dart/syntactic_scope.dart:148:5)
#4      ConstructorDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:3078:15)
#5      NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:9166:20)
#6      SyntacticScopeNamesCollector._visitClassOrMixinMembers (package:analyzer_plugin/src/utilities/change_builder/dart/syntactic_scope.dart:320:18)
#7      SyntacticScopeNamesCollector.visitClassDeclaration (package:analyzer_plugin/src/utilities/change_builder/dart/syntactic_scope.dart:126:5)
#8      ClassDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:1765:49)
#9      NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:9166:20)
#10     CompilationUnitImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:2565:21)
#11     RecursiveAstVisitor.visitCompilationUnit (package:analyzer/dart/ast/visitor.dart:780:10)
#12     CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2558:49)
#13     importLibraryElementImpl (package:analyzer_plugin/src/utilities/change_builder/dart/import_library_element.dart:58:25)
#14     DartFileEditBuilderImpl.importLibraryElement (package:analyzer_plugin/src/utilities/change_builder/change_builder_dart.dart:1219:19)
#15     CompletionDomainHandler.getSuggestionDetails.<anonymous closure> (package:analysis_server/src/domain_completion.dart:184:28)
#16     DartChangeBuilderImpl.addFileEdit.<anonymous closure> (package:analyzer_plugin/src/utilities/change_builder/change_builder_dart.dart:55:20)
#17     ChangeBuilderImpl.addFileEdit (package:analyzer_plugin/src/utilities/change_builder/change_builder_core.dart:73:18)
<asynchronous suspension>
#18     DartChangeBuilderImpl.addFileEdit (package:analyzer_plugin/src/utilities/change_builder/change_builder_dart.dart:52:18)
#19     CompletionDomainHandler.getSuggestionDetails (package:analysis_server/src/domain_completion.dart:183:19)
<asynchronous suspension>
#20     CompletionDomainHandler.handleRequest.<anonymous closure> (package:analysis_server/src/domain_completion.dart:210:9)
#21     _rootRun (dart:async/zone.dart:1124:13)
#22     _CustomZone.run (dart:async/zone.dart:1021:19)
#23     _runZoned (dart:async/zone.dart:1516:10)
#24     runZoned (dart:async/zone.dart:1500:12)
#25     CompletionDomainHandler.handleRequest (package:analysis_server/src/domain_completion.dart:206:12)
#26     AnalysisServer.handleRequest.<anonymous closure>.<anonymous closure> (package:analysis_server/src/analysis_server.dart:349:45)
#27     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)
#28     AnalysisServer.handleRequest.<anonymous closure> (package:analysis_server/src/analysis_server.dart:345:50)
#29     _rootRun (dart:async/zone.dart:1124:13)
#30     _CustomZone.run (dart:async/zone.dart:1021:19)
#31     _runZoned (dart:async/zone.dart:1516:10)
#32     runZoned (dart:async/zone.dart:1500:12)
#33     AnalysisServer.handleRequest (package:analysis_server/src/analysis_server.dart:344:5)
#34     ByteStreamServerChannel._readRequest.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:174:16)
#35     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)
#36     ByteStreamServerChannel._readRequest (package:analysis_server/src/channel/byte_stream_channel.dart:165:47)
#37     ByteStreamServerChannel.listen.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:110:26)
#38     _rootRunUnary (dart:async/zone.dart:1132:38)
#39     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#40     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#41     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#42     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#43     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:68:11)
#44     _EventSinkWrapper.add (dart:async/stream_transformers.dart:15:11)
#45     _StringAdapterSink.add (dart:convert/string_conversion.dart:238:11)
#46     _LineSplitterSink._addLines (dart:convert/line_splitter.dart:150:13)
#47...
```

For additional log information, please append the contents of
file:///private/var/folders/rl/2tj3lftd7cz5r2z_s6h0w0f00000gn/T/report.txt.

**Comment by srawlins** [comment]
Thanks for reporting this issue!

Dart 2.2 is several months old now; newer versions of Dart (and Flutter) may have fixed this issue. Try upgrading Dart (or Flutter if you are using Flutter).

Additionally, can you show the content of the file you were working on when this error occurred? Or perhaps just the constructor with the `@required this.url` bit?

**Comment by MichaelRFairhurst** [comment]
I think this has been fixed @scheglov

**Comment by scheglov** [comment]
Was fixed in d1b15cc92c883479108fe29988f89a37a187ce4b
